torch/_inductor/codegen/wrapper.py

```diff
@@ -3833,3 +3838,16 @@ def write_get_raw_stream_header_once(self) -> None:
         #         V.graph.device_ops.import_get_raw_stream_as("get_raw_stream")
         #     )
         self.parent_wrapper.write_get_raw_stream_header_once()
+
+    @cache_on_self
+    def get_root_graph(self) -> PythonWrapperCodegen:
+        root: PythonWrapperCodegen | SubgraphPythonWrapperCodegen = self
+        while isinstance(root, SubgraphPythonWrapperCodegen):
+            root = root.parent_wrapper
+
+        assert isinstance(root, PythonWrapperCodegen)
+        return root
+
+    def generate_and_run_autotune_block(self):
```

Comment:
I'm a bit confused by this. why do we not want to do auto-tuning block in subgraph? wouldnt there be kernels that might be in the subgraph but not parent?

Reply:
Here we are talking about inductor-partitioned subgraphs, so there is no control flow, so all the subgraphs will be executed, which means all the kernels will be executed. Also all the kernels are defined in the global scope anyway.

This is also relevant to the autotune_at_compile_time discussion. If that's turned off, we don't really need this PR.